### DISCUSSION TITLE
Pause indexing?

### DISCUSSION THREAD

**Discussion by tonymurray** [photoprism/photoprism#3174] [Q&A]
Hi ,

I am currently re-indexing my collection after reinstall on a NAS device. The last time I indexed it took several days to complete.
Since I normally shutdown the NAS on a nightly basis: is it possible to pause indexing? 
If the computer is shutdown, will indexing need to be re-started from scratch?

Thanks for info!

**Comment by graciousgrey** [comment]
There is no pause functionality at the moment only cancel (stop). So yes, you would need to trigger indexing again after restarting the app.
The normal indexing skips files that have not been updated and are already indexed. The [complete indexing](https://docs.photoprism.app/user-guide/library/originals/#when-should-complete-rescan-be-selected)  indexes every file.
When you are doing a complete rescan you could do it folder wise (you can select the folder  to index in the web ui).

**Reply by thitiv** [reply]
I am indexing my whole QNAP NAS as well. It contains 500,000+ pictures and it has been taking weeks. My first round I installed PhotoPrism on the NAS and it consumed so much CPU power that I decided to switch to PhotoPrism on Docker on a Core i9 Mac with the NAS mounted as network drive. 

Never completed yet because either 
(1) either the NAS or the Mac need to restart during the weeks and I had to start over again after that; OR
(2) the mounted network drive is disconnected from a network glitch that sometimes occur, and PhotoPrism thought the mapped drive was removed, put lost photos as hidden which I can’t find how to restore and have to start indexing all over again.

Appreciate advices. Would split the folders into smaller ones and index sub folders gradually help? But would it solve my problem number (2)?

**Reply by graciousgrey** [reply]
Restarting should not be an issue, if your are running a [normal index (not a complete rescan )](https://docs.photoprism.app/user-guide/library/originals/#when-should-complete-rescan-be-selected). If you start indexing again after a restart of your Mac it will proceed indexing not yet indexed or updated files. It won't index already indexed, unchanged files again. 

What version of PhotoPrism are you running? We have implemented a check that prevents PhotoPrism from removing files from the index if the originals folder is completely empty e.g. due to network issues.

- https://github.com/photoprism/photoprism/issues/2299
